src/demo/java/demo/customplugins/README.md

```diff
@@ -0,0 +1,390 @@
+# AWS JDBC Driver for MySQL Custom Connection Plugin Sample Application
+
+This README walks through the steps to create 2 additional custom connection plugins for the AWS
+JDBC Driver for MySQL and use them in a simple JDBC application. All the code used in this README
+are available within this directory, to run the sample application directly,
+see [Running the Sample Code](#running-the-sample-code) section.
+
+### Minimum Requirements
+
+To following along this tutorial or to run this sample program directly, the following are required:
+
+1. Amazon Corretto 8+ or Java 8+
+2. A MySQL database
+3. The AWS JDBC Driver for MySQL must be added to the classpath. To obtain the AWS JDBC Driver for
+   MySQL
+   see [here](https://github.com/awslabs/aws-mysql-jdbc#obtaining-the-aws-jdbc-driver-for-mysql).
+
+## What are Connection Plugins
+
+Connection plugins are widgets attached to each `Connection` objects to help execute SQL functions
+related to that `Connection`. All the connection plugins are chained together, where the prior
+connection plugin calls the next plugin. The AWS JDBC Driver for MySQL provides 2 default connection
+plugins, the `NodeMonitoringConnectionPlugin` and the `DefaultConnectionPlugin`.
```

Comment:
I wouldn't mention NodeMonitoringConnectionPlugin here.

Reply:
DefaultConnectionPlugin is added to the tail of plugin chain by the driver and actually executes a method.

Reply:
Need to add few words about importance of plugins in a chain.